storage/src/main/java/fk/prof/storage/StorageBackedOutputStream.java

```diff
@@ -0,0 +1,108 @@
+package fk.prof.storage;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+
+/**
+ * OutputStream implementation which is backed by a {@link AsyncStorage}.
+ * Every {@code partSize} chunk of the bytes will be written to the storage.
+ * The target path of the chunk will be decided by the {@link FileNamingStrategy}.
+ *
+ * @see StorageBackedInputStream
+ * @author gaurav.ashok
+ */
+public class StorageBackedOutputStream extends OutputStream {
```

Comment:
Assuming this class doesn't need to be thread-safe. Confirm if this is a valid assumption.